retry.go

```diff
@@ -0,0 +1,64 @@
+package hutils
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+const (
+	DefaultRTimes = 5
+	DefaultRDelay = 3000 * time.Millisecond
+)
+
+type RetryConfig struct {
+	context  context.Context
+	times    uint
+	duration time.Duration
+}
+
+type RetryFunc func() error
+
+type Option func(*RetryConfig)
+
+func RTimes(n uint) Option {
+	return func(retryConfig *RetryConfig) {
+		retryConfig.times = n
+	}
+}
+
+func RDelay(d time.Duration) Option {
+	return func(retryConfig *RetryConfig) {
+		retryConfig.duration = d
+	}
+}
+
+func Context(ctx context.Context) Option {
+	return func(rc *RetryConfig) {
+		rc.context = ctx
+	}
+}
+
+func Retry(retryFunc RetryFunc, opt ...Option) error {
```

Comment:
it seems that use a `ctx` param is more straightforward than use  a`Context ` option?